components/brave_education/resources/brave_education.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { EducationPageProxy } from './education_page_proxy'
+import { readCommandMessage } from './message_data_reader'
+
+function pageReady() {
+  return new Promise<void>((resolve) => {
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', () => resolve())
+    } else {
+      resolve()
+    }
+  })
+}
+
+async function handleMessage(data: any) {
+  // Read the message provided by the website. If the message is invalid, then
+  // an error will be thrown and will appear on the console for debugging.
+  const message = readCommandMessage(data)
+
+  // Execute the command.
+  const { handler } = EducationPageProxy.getInstance()
+  const { executed } = await handler.executeCommand(message.command);
+  if (!executed) {
+    console.warn(`Command "${ message.command }" could not be executed`)
+  }
+}
+
+async function main() {
+  await pageReady()
+
+  // Get the website URL that corresponds to this webUI URL. If there is no
+  // match, then do not load anything into the iframe and leave the page empty.
+  const { handler } = EducationPageProxy.getInstance()
+  const { serverUrl } = await handler.getServerUrl()
+  if (!serverUrl) {
+    console.warn('No matching server URL')
+    return
+  }
+
+  // Load the education content into the iframe.
+  const frame = document.getElementById('content') as HTMLIFrameElement
+  frame.src = serverUrl
+
+  // Attach a postMessage handler for messages originating from the iframe.
+  window.addEventListener('message', (event) => {
+    if (event.origin === 'https://brave.com' ||
+        event.origin === 'chrome://webui-test') {
```

Comment:
q: what's this origin used for?

Reply:
This origin is used by the browser tests in `education_page_ui_browsertest.cc`.

Reply:
Sounds good. To me it seems odd to grant this privilege since it's only for testing.

nit: if it's easy to change, it would be useful to only go down this code path if there's a test flag or something like that enabled (and available), otherwise feel free to ignore this one.

Not a blocker